### PR TITLE
Add global option for MPAS driver

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Synopsis**

Fixes Issue #659 

**Type**

- [x] Bug fix (corrects a known issue)
- [ ] Code maintenance (refactoring, etc. without behavior change)
- [x] Documentation
- [ ] Enhancement (adds new functionality)
- [ ] Tooling (CI, code-quality, packaging, revision-control, etc.)

**Impact**

- [ ] This is a breaking change (changes existing functionality)
- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: docs/sections/user_guide/yaml/components/mpas.rst [hunk 1]
```diff
@@ -17,15 +17,20 @@ An MPAS build provides prototype versions of certain required runtime files. Her
...
+Accepted values are ``global`` and ``regional``.
+
 execution:
 ^^^^^^^^^^
 
 See :ref:`this page <execution_yaml>` for details.
 
-boundary_conditions:
-^^^^^^^^^^^^^^^^^^^^
+lateral_boundary_conditions:
```
Comment (christinaholtNOAA) [inline docs/sections/user_guide/yaml/components/mpas.rst:30]: This was a copy/paste issue with MPASInit docs, where MPASInit combines initial and lateral boundary conditions, and MPAS does not.

File: docs/sections/user_guide/yaml/components/mpas_init.rst [hunk 2]
```diff
@@ -43,6 +38,11 @@ Describes the boundary condition files needed for the forecast. These will most
 
 An absolute path to the output of the ``ungrib`` tool that will be used to prepare MPAS-ready initial and lateral boundary conditions. The names of the files are specified in the ``streams.init_atmosphere`` XML file, and may be specified in the ``streams: values:`` block of the driver YAML.
 
+execution:
+^^^^^^^^^^
+
+See :ref:`this page <execution_yaml>` for details.
+
```
Comment (christinaholtNOAA) [inline docs/sections/user_guide/yaml/components/mpas_init.rst:45]: Alphabetizing.

File: src/uwtools/drivers/mpas.py [hunk 3]
```diff
@@ -66,6 +66,23 @@ def namelist_file(self):
...
+            self.files_copied(),
+            self.files_linked(),
+            self.namelist_file(),
+            self.runscript(),
+            self.streams_file(),
+        ]
+        if self.config["domain"] == "regional":
+            required.append(self.boundary_files())
+        yield required
+
```
Comment (christinaholtNOAA) [inline src/uwtools/drivers/mpas.py:85]: MPAS and MPASInit now have their own `provisioned_rundir` methods and the MPASBase class has only the interface definition.

**Review by christinaholtNOAA — COMMENTED** [review]
Most of the changes here follow how FV3 has implemented the need for global/regional domains and the dependence on lateral boundary conditions for the regional domain.

**Review by elcarpenterNOAA — APPROVED** [review]

**Review by nbharwani-cu — APPROVED** [review]

**Review by maddenp-cu — APPROVED** [review]
Nice.

**Review by Byrnetp — APPROVED** [review]

**Pull request merged**